File: frontend/src/app/core/services/store-settings-service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StoreSettingsRequestDto, StoreSettingsResponseDto } from '../models/store-settings.model';

@Injectable({
  providedIn: 'root',
})
export class StoreSettingsService {
  private http = inject(HttpClient);
  private readonly endPoint = 'StoreSettings';

  getSettings(): Observable<StoreSettingsResponseDto> {
    return this.http.get<StoreSettingsResponseDto>(this.endPoint);
  }

  updateSettings(dto: StoreSettingsRequestDto): Observable<StoreSettingsResponseDto> {
    return this.http.put<StoreSettingsResponseDto>(this.endPoint, dto);
  }

  uploadLogo(file: File, isBig: boolean): Observable<{ url: string }> {
    const formData = new FormData();
    formData.append('file', file);
    return this.http.post<{ url: string }>(`${this.endPoint}/upload-logo?isBig=${isBig}`, formData);
  }
}

File: backend/RadianciaKS.Application/Services/StoreSettingsService.cs
using FluentValidation;
using Microsoft.AspNetCore.Http;
using Microsoft.EntityFrameworkCore;
using RadianciaKS.Application.DTOs.StoreSettings;
using RadianciaKS.Application.Interfaces;
using RadianciaKS.Application.Mappers;
using RadianciaKS.Application.Services.Interfaces;
using RadianciaKS.Domain.Models;

namespace RadianciaKS.Application.Services
{
    public class StoreSettingsService : IStoreSettingsService
    {
        private readonly IApplicationDbContext _context;
        private readonly IValidator<StoreSettingsRequestDto> _validator;
        private readonly IImageStorageService _imageStorageService;
        private readonly StoreSettingsMapper _mapper;

        public StoreSettingsService(IApplicationDbContext applicationDbContext, IValidator<StoreSettingsRequestDto> validator, IImageStorageService imageStorageService)
        {
            _context = applicationDbContext;
            _validator = validator;
            _imageStorageService = imageStorageService;
            _mapper = new StoreSettingsMapper();
        }

        public async Task<StoreSettingsResponseDto> GetSettings()
        {
            var settings = await _context.StoreSettings.FirstOrDefaultAsync();
            if (settings == null)
            {
                settings = new StoreSettings { StoreName = "Novo Restaurante" };
                _context.StoreSettings.Add(settings);
                await _context.SaveChangesAsync();
            }
            return _mapper.ToDto(settings);
        }

        public async Task<StoreSettingsResponseDto> UpdateSettings(StoreSettingsRequestDto dto)
        {
            await _validator.ValidateAndThrowAsync(dto);

            var storeSettings = await _context.StoreSettings.FirstOrDefaultAsync();

            if (storeSettings == null)
                throw new ArgumentException("A loja informada não existe.");

            storeSettings.Update(
                dto.StoreName,
                dto.CNPJ,
                dto.Address,
                dto.Phone,
                dto.ReceiptFooter,
                dto.SmallLogoPath,
                dto.BigLogoPath,
                dto.ServiceCharge
            );

            await _context.SaveChangesAsync();
            return _mapper.ToDto(storeSettings);
        }

        public async Task<string> UploadLogo(IFormFile file, bool isBig)
        {
            var storeSettings = await _context.StoreSettings.FirstOrDefaultAsync();
            if (storeSettings == null)
                throw new ArgumentException("A loja informada não existe.");

            var suffix = isBig ? "big" : "small";

            var imageUrl = await _imageStorageService.UploadImageAsync(file, $"logo_{suffix}");

            if (!isBig)
                storeSettings.SmallLogoPath = imageUrl;
            else
                storeSettings.BigLogoPath = imageUrl;

            await _context.SaveChangesAsync();

            return imageUrl;
        }
    }
}